#<C>Class Project</C>

##Codebook
After a review of the various code books available for the class, I decided to use the Gapminder data set.

##Research Question
Is there an association between Internet usage and suicide rates?  I will use the variables Internetuserate and suicideper100TH rate in my code book.  

##Data

The mission of Gapminder Foundation is to fight ignorance with a fact-based worldview that everyone can understand. They use surveys to ask representative groups of people simple questions about key-aspects of global development.  When they find large knowledge-gaps, they know what teaching materials we should develop. Gapminder Foundation is a non-profit organization promoting sustainable global development and achievement of the United Nations Millennium Development Goals by increased use and understanding of statistics and other information about social, economic and environmental development at local, national and global levels.  Gapminder was founded in Stockholm by Ola Rosling, Anna Rosling Rönnlund and Hans Rosling on February 25, 2005. 

Gapminder's data is released under the Creative Commons Attribution 3.0 Unported license. 

###Sources of Data for Variables

<I>Internet Use per 100</I> from the [world bank](http://data.worldbank.org/indicator/IT.NET.USER.P2).  The World Bank defined Internet users as individuals who have used the Internet (from any location) in the last 12 months. Internet can be used via a computer, mobile phone, personal digital assistant, games machine, digital TV etc.  

<I>Suicide, age adjusted, per 100 000</I> from the [World Health Organization] (https://www.google.com/url?q=http://www.who.int/violence_injury_prevention/surveillance/databases/mortality/en/index.html&sa=D&ust=1448252492492000&usg=AFQjCNFlYUy92km_25IBO4OsX53Wp_U4sQ). WHO defined Suicide as mortality due to self-inflicted injury, per 100 000 standard population, age adjusted.

<I>Income per person.</I> Gapminder complied income per person from several sources.  

##Literature Review
There is concern that the Internet is playing an increasing role in self-harm and suicide. Internet use may exert both positive and negative effects on young people at risk of self-harm or suicide. The internet has created channels of communication that can be misused to 'cyber-bully' peers; both cyber-bullying and general internet use have been found to correlate with increased risk of self-harm, suicidal idealization, and depression.  Youth who self-harm or are suicidal often make use of the internet. It is most commonly used for constructive reasons such as seeking support and coping strategies, but may exert a negative influence, normalizing self-harm and potentially discouraging disclosure or professional help-seeking. Correlations have been found between internet exposure and violent methods of self-harm.

Baker D, & Fortune S (2008) Understanding self-harm and suicide websites a qualitative interview study of young adult website users. Crisis 29: 118-122.

Daine K, Hawton K, Singaravelu V, Stewart A, Simkin S, Montgomery P (2013) The Power of the Web: A Systematic Review of Studies of the Influence of the Internet on Self-Harm and Suicide in Young People. PLoS ONE 8(10): e77555. doi:10.1371/journal.pone.0077555

Dunlop SM, More E, Romer D (2011) Where do youth learn about suicides on the Internet, and what influence does this have on suicidal ideation? J Child Psychol Psychiatry 52: 1073-1080.

Durkee T, Hadlaczky G, Westerlund M, Carli V (2011) Internet pathways in suicidality: A review of the evidence. IntJEnvironResPublic Health 8: 3938- 3952.

Gould MS, Wallenstein S, Kleinman MH, O'Carroll P, Mercy J (1990) Suicide clusters: an examination of age-specific effects. Am J Public Health 80: 211-212.

Hawton K, Saunders KEA, O'Connor RC (2012) Self-harm and suicide in adolescents. Lancet 379: 2373-2382.

Kim K, Ryu E, Chon MY, Yeun EJ, Choi SY, et al. (2006) Internet addiction in Korean adolescents and its relation to depression and suicidal ideation: A questionnaire survey. Int J Nurs Stud 43: 185-192.

Katsumata Y, Matsumoto T, Kitani M, Takeshima T (2008) Electronic media use and suicidal ideation in Japanese adolescents. Psychiatry Clin Neurosci 62: 744-746.

Madge N, Hawton K, McMahon EM, De Leo D, de Wilde EJ, et al. (2011) Psychological characteristics, stressful life events and deliberate self-harm: findings from the Child & Adolescent Self-harm in Europe (CASE) Study. Eur Child Adolesc Psychiatry 20: 499-508.

O'Connor RC, Rasmussen S, Hawton K (2012) Distinguishing adolescents who think about self-harm from those who engage in self-harm. Br J Psychiatry 200: 330-335.

Patton GC, Coffey C, Sawyer SM, Viner RM, Haller DM, et al. (2009) Global patterns of mortality in young people: a systematic analysis of population health data. Lancet 374: 881-892.

##Hypothesis
There is a positive relationship between Internet usage rates and suicide rates.

##Data
Before beginning any analysis, I needed to import some python libraries.

In [7]:
#import graphlab as gl
import numpy as np
print ("numpy version: ", np.__version__)
import sys
print (sys.version)
import pylab
import matplotlib.pyplot as plt
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf
import statsmodels.stats.multicomp as multi 
#import graphlab.numpy
import seaborn
import scipy

#gl.canvas.set_target('ipynb') # Print graphlab graphs in line
%matplotlib inline

numpy version:  1.10.1
3.5.1 |Anaconda 2.4.1 (64-bit)| (default, Jan 19 2016, 12:15:43) [MSC v.1900 64 bit (AMD64)]


I downloaded the data in a CSV file (gapminder.csv) from the course website and saved it to D:/Data/CourseraClass directory on my machine.  I loaded the data in Python from the CSV file. 

In [8]:
data = pd.read_csv('D:/Data/CourseraClass/gapminder.csv')
len(data)

213

## Exploratory Analysis
Review the top of the data.

In [9]:
data.head()

country   incomeperperson alcconsumption armedforcesrate  \
0  Afghanistan                              .03        .5696534   
1      Albania  1914.99655094922           7.29       1.0247361   
2      Algeria  2231.99333515006            .69        2.306817   
3      Andorra  21943.3398976022          10.17                   
4       Angola  1381.00426770244           5.57       1.4613288   

  breastcancerper100th      co2emissions  femaleemployrate hivrate  \
0                 26.8          75944000  25.6000003814697           
1                 57.4  223747333.333333  42.0999984741211           
2                 23.5  2932108666.66667  31.7000007629394      .1   
3                                                                    
4                 23.1         248358000  69.4000015258789       2   

    internetuserate lifeexpectancy     oilperperson polityscore  \
0  3.65412162280064         48.673                            0   
1  44.9899469578783         76.918                            9   
2  12.5000733055148         73.131  .42009452521537           2   
3                81                                               
4  9.99995388324075         51.093                           -2   

  relectricperperson   suicideper100th        employrate urbanrate  
0                     6.68438529968262  55.7000007629394     24.04  
1   636.341383366604  7.69932985305786  51.4000015258789     46.72  
2   590.509814347428   4.8487696647644              50.5     65.22  
3                     5.36217880249023                       88.92  
4   172.999227388199  14.5546770095825  75.6999969482422      56.7

Review the bottom of the data.

In [10]:
data.tail()

country   incomeperperson alcconsumption armedforcesrate  \
208             Vietnam  722.807558834445           3.91       1.0853671   
209  West Bank and Gaza                                        5.9360854   
210         Yemen, Rep.    610.3573673206             .2       2.3162346   
211              Zambia  432.226336974583           3.56        .3413352   
212            Zimbabwe  320.771889948584           4.96       1.0327854   

    breastcancerper100th      co2emissions  femaleemployrate hivrate  \
208                 16.2        1425435000  67.5999984741211      .4   
209                       14241333.3333333  11.3000001907349           
210                 35.1  234864666.666667  20.2999992370605           
211                   13  132025666.666667              53.5    13.5   
212                   19  590219666.666666  58.0999984741211    14.3   

      internetuserate lifeexpectancy oilperperson polityscore  \
208  27.8518215557703         75.181                       -7   
209  36.4227717919075         72.832                            
210  12.3497504635596         65.493                       -2   
211   10.124986462443         49.025                        7   
212  11.5004153209768         51.384                        1   

    relectricperperson   suicideper100th        employrate urbanrate  
208   302.725654656034  11.6533222198486                71     27.84  
209                                                     32      71.9  
210    130.05783139719  6.26578903198242                39     30.64  
211   168.623030511023  12.0190362930298                61     35.42  
212   297.883200408304  13.9052667617798  66.8000030517578     37.34

Subset the data to remove missing values from the variables of interest.

In [1]:
data['suicideper100th'] = data['suicideper100th'].replace(r'\s+', np.nan, regex=True)
data['internetuserate'] = data['internetuserate'].replace(r'\s+', np.nan, regex=True)
data['incomeperperson']= data['incomeperperson'].replace(r'\s+', np.nan, regex=True)
sub = data.dropna(subset = ['suicideper100th', 'internetuserate', 'incomeperperson'])
sub1 = sub[['suicideper100th', 'internetuserate', 'incomeperperson']]
len(sub1)
len(sub)

NameError: name 'data' is not defined

Convert fields to numeric

In [12]:
sub1['suicideper100th'] = sub1['suicideper100th'].convert_objects(convert_numeric=True)
sub1['internetuserate'] = sub1['internetuserate'].convert_objects(convert_numeric=True)
sub1['incomeperperson'] = sub1['incomeperperson'].convert_objects(convert_numeric=True)

C:\Users\bryan_000\Anaconda3\lib\site-packages\ipykernel\__main__.py:1: FutureWarning: convert_objects is deprecated.  Use the data-type specific converters pd.to_datetime, pd.to_timedelta and pd.to_numeric.
  if __name__ == '__main__':
C:\Users\bryan_000\Anaconda3\lib\site-packages\ipykernel\__main__.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if __name__ == '__main__':
C:\Users\bryan_000\Anaconda3\lib\site-packages\ipykernel\__main__.py:2: FutureWarning: convert_objects is deprecated.  Use the data-type specific converters pd.to_datetime, pd.to_timedelta and pd.to_numeric.
  from ipykernel import kernelapp as app
C:\Users\bryan_000\Anaconda3\lib\site-packages\ipykernel\__main__.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a sl

Create categorial predictor based on percentiles.  Those in the lowest 25th percentile will be coded 1 to indicate low usage. Those in the upper 75th percentile will be coded as 3 to indicate high usage.  Those in between will be coded as 2 to indicate medium usage.

In [13]:
print (round(np.percentile(sub1['internetuserate'],25)))
print (round(np.percentile(sub1['internetuserate'],50)))
print (round(np.percentile(sub1['internetuserate'],75)))

10.0
29.0
52.0


In [14]:
sub1['internetrate'] = 0
sub1['internetrate'][(sub1['internetuserate']>52)] = 3
sub1['internetrate'][(sub1['internetuserate']<10)] = 1
sub1['internetrate'][(sub1['internetuserate']>=10) & (sub1['internetuserate']<=52)] = 2

C:\Users\bryan_000\Anaconda3\lib\site-packages\ipykernel\__main__.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if __name__ == '__main__':
C:\Users\bryan_000\Anaconda3\lib\site-packages\ipykernel\__main__.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  from ipykernel import kernelapp as app
C:\Users\bryan_000\Anaconda3\lib\site-packages\pandas\core\generic.py:4059: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._upd

Create categorial response based on percentiles. Those in the lower 50 percentile will be coded 1 to indicate low rate. Those in the upper 50 percentile will be coded as 2 to indicate high rate.

In [15]:
print (round(np.percentile(sub1['suicideper100th'],25)))
print (round(np.percentile(sub1['suicideper100th'],50)))
print (round(np.percentile(sub1['suicideper100th'],75)))

5.0
8.0
12.0


In [16]:
sub1['suiciderate'] = 0
sub1['suiciderate'][(sub1['suicideper100th']>=8)] = 2
sub1['suiciderate'][(sub1['suicideper100th']< 8)] = 1

C:\Users\bryan_000\Anaconda3\lib\site-packages\ipykernel\__main__.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if __name__ == '__main__':
C:\Users\bryan_000\Anaconda3\lib\site-packages\ipykernel\__main__.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  from ipykernel import kernelapp as app
C:\Users\bryan_000\Anaconda3\lib\site-packages\pandas\core\generic.py:4059: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._upd

Create categorial moderator based on percentiles.  Those in the lowest 25th percentile will be coded 1 to indicate low income. Those in the upper 75th percentile will be coded as 3 to indicate high income.  Those in between will be coded as 2 to indicate medium income.

In [17]:
print (round(np.percentile(sub1['incomeperperson'],25)))
print (round(np.percentile(sub1['incomeperperson'],50)))
print (round(np.percentile(sub1['incomeperperson'],75)))

726.0
2454.0
8572.0


In [18]:
sub1['ipp'] = 0
sub1['ipp'][(sub1['incomeperperson']>8572.0)] = 3
sub1['ipp'][(sub1['incomeperperson']<726)] = 1
sub1['ipp'][(sub1['incomeperperson']>=726) & (sub1['incomeperperson']<=8572)] = 2

C:\Users\bryan_000\Anaconda3\lib\site-packages\ipykernel\__main__.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if __name__ == '__main__':
C:\Users\bryan_000\Anaconda3\lib\site-packages\ipykernel\__main__.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  from ipykernel import kernelapp as app
C:\Users\bryan_000\Anaconda3\lib\site-packages\pandas\core\generic.py:4059: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._upd

Create categorial moderator based on percentiles.  Those in the lowest 50th percentile will be coded 1 to indicate low income. Those in the upper 50th percentile will be coded as 2 to indicate high income.  

In [20]:
sub1['ipp2'] = 0
sub1['ipp2'][(sub1['incomeperperson']>=2454)] = 2
sub1['ipp2'][(sub1['incomeperperson']< 2454)] = 1

C:\Users\bryan_000\Anaconda3\lib\site-packages\ipykernel\__main__.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if __name__ == '__main__':
C:\Users\bryan_000\Anaconda3\lib\site-packages\ipykernel\__main__.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  from ipykernel import kernelapp as app
C:\Users\bryan_000\Anaconda3\lib\site-packages\pandas\core\generic.py:4059: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._upd

Delete blanks

In [21]:
sub2 = sub1[ (sub1['internetrate'] > 0) & (sub1['suiciderate']  > 0) & (sub1['ipp']  > 0) ]
len(sub2)

174

##Numerical Summaries

Compute mean and standard deviation for Internet Usage Rate (Independent Variable)

In [22]:
print (round(sub2['internetuserate'].mean(),2))
print (round(sub2['internetuserate'].std(),2))

33.97
27.67


Compute mean and standard deviation for Suicide Rate (Dependent Variable)

In [23]:
print (round(sub2['suicideper100th'].mean(),2))
print (round(sub2['suicideper100th'].std(),2))

9.71
6.3


Compute mean and standard deviation for Income per Person (Potential Moderator)

In [24]:
print (round(sub2['incomeperperson'].mean(),2))
print (round(sub2['incomeperperson'].std(),2))

7165.31
10229.44


Convert new column to numeric

In [25]:
#del sub2['internetuserate']
sub2['internetrate']= sub2['internetrate'].convert_objects(convert_numeric=True)
sub2.head()

C:\Users\bryan_000\Anaconda3\lib\site-packages\ipykernel\__main__.py:2: FutureWarning: convert_objects is deprecated.  Use the data-type specific converters pd.to_datetime, pd.to_timedelta and pd.to_numeric.
  from ipykernel import kernelapp as app


suicideper100th  internetuserate  incomeperperson  internetrate  \
1         7.699330        44.989947      1914.996551             2   
2         4.848770        12.500073      2231.993335             2   
3         5.362179        81.000000     21943.339898             3   
4        14.554677         9.999954      1381.004268             1   
5         2.161843        80.645455     11894.464075             3   

   suiciderate  ipp  ipp2  
1            1    2     1  
2            1    2     1  
3            1    3     2  
4            2    2     1  
5            1    3     2

## Anova Model
Using ols function for calculating the F-statistic and associated p value

In [ ]:
model1=smf.ols(formula='suicideper100th ~ C(internetrate)', data=sub2)
results1 = model1.fit()
print (results1.summary())

##Model Interpretation for ANOVA:

When examining the association between suicide rate (quantitative response) and internet usage rate, an Analysis of Variance (ANOVA) revealed that suicide rate (Mean=9.7, s.d. ±6.3) was not related to Internet Usage Rate (Mean=33.97, s.d. ±27.7), F(1, 171)=.8787, p>.05.

Compute group means & group standard deviations

In [ ]:
sub3 = sub2[['suicideper100th', 'internetrate']]
print ('Means for Suicide Rates by Internet Use')
m2= sub3.groupby('internetrate').mean()
print (m2)

In [ ]:
print ('Standard deviations for Suicide Rates by Internet Use')
sd2 = sub3.groupby('internetrate').std()
print (sd2)

###Post Hoc Analysis
Although significant results were not established, a post hoc analysis was conducted to demonstrate ability to run post hoc analysis.

In [ ]:
mc1 = multi.MultiComparison(sub2['suicideper100th'], sub2['internetrate'])
res1 = mc1.tukeyhsd()
print(res1.summary())

###Scatter Plot

In [ ]:
scat1 = seaborn.regplot(x="internetuserate", y="suicideper100th", fit_reg=True, data=sub2)
plt.xlabel('Internet Use Rate')
plt.ylabel('Suicide Rate')
plt.title('Scatterplot for the Association Between Internet Use Rate and Suicide Rate')

##Correlation

In [ ]:
print ('Association between Internet Use Rate and Suicide Rate')
r = scipy.stats.pearsonr(sub2['internetuserate'], sub2['suicideper100th'])
print r

In [ ]:
r_squared = r[0] * r[0]
print('R Squared = '+str(r_squared))

Pearson R was .06 indicating a weak association.  The result was not statistically significant.  R-squared indicated less that 1% of the variation in suicide rates was explained by internet usage rates.

##Chi-square

In [ ]:
# contingency table of observed counts
ct1=pd.crosstab(sub2['internetrate'], sub2['suiciderate'])
print (ct1)

In [ ]:
# column percentages
colsum=ct1.sum(axis=0)
colpct=ct1/colsum
print(colpct)

In [ ]:
# chi-square
print ('chi-square value, p value, expected counts')
cs1= scipy.stats.chi2_contingency(ct1)
print (cs1)

###Model Interpretation for Chi-square:

When examining the association between suicide rate (coded as high or low) and internet usage rate (coded as high, medium or low) , a Chi Square test concluded that suicide rate was not related to Internet Usage Rate, p > .05.

### Bar Plot of Results

In [ ]:
# set rate to category variable type
sub2["internetrate"] = sub2["internetrate"].astype('category')

In [ ]:
seaborn.factorplot(x="suiciderate", y="internetrate", data=sub2, kind="bar", ci=None)
plt.xlabel('Suicide Rate ')
plt.ylabel('Internet Use (1=Low, 2=Medium, 3=High)')

###Post Hoc Analysis
Although significant results were not established, a post hoc analysis was conducted to demonstrate ability to run post hoc analysis.

In [ ]:
recode2 = {1: 1, 2: 2}
sub2["COMP1v2"] = sub2["internetrate"].map(recode2)

# contingency table of observed counts
ct2=pd.crosstab(sub2['COMP1v2'], sub2['suiciderate'])
print (ct2)

# column percentages
colsum=ct2.sum(axis=0)
colpct=ct2/colsum
print(colpct)

print ('chi-square value, p value, expected counts')
cs2= scipy.stats.chi2_contingency(ct2)
print (cs2)

In [ ]:
recode3 = {1: 1, 3: 3}
sub2["COMP1v3"] = sub2["internetrate"].map(recode3)

# contingency table of observed counts
ct3=pd.crosstab(sub2['COMP1v3'], sub2['suiciderate'])
print (ct3)

# column percentages
colsum=ct3.sum(axis=0)
colpct=ct3/colsum
print(colpct)

print ('chi-square value, p value, expected counts')
cs3= scipy.stats.chi2_contingency(ct3)
print (cs3)

In [ ]:
recode4 = {2: 2, 3: 3}
sub2["COMP2v3"] = sub2["internetrate"].map(recode4)

# contingency table of observed counts
ct4=pd.crosstab(sub2['COMP2v3'], sub2['suiciderate'])
print (ct4)

# column percentages
colsum=ct4.sum(axis=0)
colpct=ct4/colsum
print(colpct)

print ('chi-square value, p value, expected counts')
cs4= scipy.stats.chi2_contingency(ct4)
print (cs4)

###Interpretation for Post-Hoc Chi-square Analysis:

When examining the association between suicide rate (coded as high or low) and internet usage rate (coded as high, medium or low) , a Chi Square test concluded that suicide rate was not related to Internet Usage Rate, p > .05.  The post hoc analysis indicated:

Cat 1 (Low) to Cat 2 (Medium) p > .05 nonsignificant results

Cat 1 (Low) to Cat 3 (High) p > .05 nonsignificant results

Cat 2 (Low) to Cat 3 (High) P < = .05 significant results

##Testing a Potential Moderator
A moderator variable is a third variable that affects the strength of the relationship between a dependent and independent variable.  I used income per person as a potential moderator for the relationship between Interenet Usage Rate and Sucicide Rates.

### Potential Moderator Correlation
Create dataframes for each income group

In [ ]:
sub5=sub2[(sub2['ipp']== 1)]
sub6=sub2[(sub2['ipp']== 2)]
sub7=sub2[(sub2['ipp']== 3)]
sub8=sub2[(sub2['ipp2']== 1)]
sub9=sub2[(sub2['ipp2']== 2)]

In [ ]:
print ('Association between Internet Usage and Suicide Rates for LOW income countries')
r1 = scipy.stats.pearsonr(sub5['internetuserate'], sub5['suicideper100th'])
print (r1)
r_squared = r1[0] * r1[0]
print('R Squared = '+str(r_squared))
print ('       ')
print ('Pearson R was .06 indicating a weak association.  The result was not statistically significant.  R-squared indicated that less 1% of the variation in suicide rates was explained by internet usage rates.')


print ('       ')
print ('Association between Internet Usage and Suicide Rates for MIDDLE income countries')
r2 = scipy.stats.pearsonr(sub6['internetuserate'], sub6['suicideper100th'])
print (r2)
r_squared = r2[0] * r2[0]
print('R Squared = '+str(r_squared))
print ('       ')
print('Pearson R was .13 indicating a weak association.  The result was not statistically significant.  R-squared indicated that less 1% of the variation in suicide rates was explained by internet usage rates.')

print ('       ')
print ('Association between Internet Usage and Suicide Rates for HIGH income countries')
r3 = scipy.stats.pearsonr(sub7['internetuserate'], sub7['suicideper100th'])
print (r2)
r_squared = r3[0] * r3[0]
print('R Squared = '+str(r_squared))
print ('       ')
print('Pearson R was .13 indicating a weak association.  The result was not statistically significant.  R-squared indicated that less 1% of the variation in suicide rates was explained by internet usage rates.')

In [ ]:
scat1 = seaborn.regplot(x="internetuserate", y="suicideper100th", data=sub5)
plt.xlabel('Internet Use Rate')
plt.ylabel('Suicide Rate')
plt.title('Scatterplot for the Association Between Internet Use Rate and Suicide Rate for LOW income countries')
print (scat1)

In [ ]:
scat2 = seaborn.regplot(x="internetuserate", y="suicideper100th", data=sub6)
plt.xlabel('Internet Use Rate')
plt.ylabel('Suicide Rate')
plt.title('Scatterplot for the Association Between Internet Use Rate and Suicide Rate for MEDIUM income countries')
print (scat2)

In [ ]:
scat3 = seaborn.regplot(x="internetuserate", y="suicideper100th", data=sub7)
plt.xlabel('Internet Use Rate')
plt.ylabel('Suicide Rate')
plt.title('Scatterplot for the Association Between Internet Use Rate and Suicide Rate for HIGH income countries')
print (scat3)

### Potential Moderator - Chi-square

In [ ]:
print ('Association between Internet Use Rate and Suicide Rate for those in LOW Income Countries')
# contingency table of observed counts
ct8=pd.crosstab(sub8['internetrate'], sub8['suiciderate'])
print (ct8)

# column percentages
colsum=ct8.sum(axis=0)
colpct=ct8/colsum
print(colpct)

In [ ]:
print ('Association between Internet Use Rate and Suicide Rate for those in HIGH Income Countries')
# contingency table of observed counts
ct9=pd.crosstab(sub9['internetrate'], sub9['suiciderate'])
print (ct9)

# column percentages
colsum=ct9.sum(axis=0)
colpct=ct9/colsum
print(colpct)

In [ ]:
# chi-square
print ('Chi-Square for those in LOW Income Countries')
print ('chi-square value, p value, expected counts')
cs8= scipy.stats.chi2_contingency(ct8)
print (cs8)

In [ ]:
# chi-square
print ('Chi-Square for those in HIGH Income Countries')
print ('chi-square value, p value, expected counts')
cs9= scipy.stats.chi2_contingency(ct9)
print (cs9)

###Model Interpretation for Potential Moderator Chi-square:

When examining the association between suicide rate (coded as high or low) and internet usage rate (coded as high, medium or low) using Personal income as, a moderator, the Chi Square test concluded that suicide rate was not related to Internet Usage Rate, p > .05, regardless of the income level.

###Visual Displays

In [ ]:
seaborn.factorplot(x="suiciderate", y="internetrate", data=sub8, kind="bar", ci=None)
plt.xlabel('Suicide Rate ')
plt.ylabel('Internet Use (1=Low, 2=Medium, 3=High)')
plt.title('Association between Internet Usage Rate and Suicide Rate for those with LOW income')

In [ ]:
seaborn.factorplot(x="suiciderate", y="internetrate", data=sub9, kind="bar", ci=None)
plt.xlabel('Suicide Rate ')
plt.ylabel('Internet Use (1=Low, 2=Medium, 3=High)')
plt.title('Association between Internet Usage Rate and Suicide Rate for those with HIGH income')

In [ ]:
seaborn.factorplot(x='internetrate', y='suiciderate', data=sub8, kind="point", ci=None)
plt.xlabel('Internet Usage (Low, Medium, High)')
plt.ylabel('Suicide Rate')
plt.title('Association between Internet Usage Rate and Suicide Rate for those with LOW income')

seaborn.factorplot(x="internetrate", y="suiciderate", data=sub9, kind="point", ci=None)
plt.xlabel('Internet Usage (Low, Medium, High)')
plt.ylabel('Suicide Rate')
plt.title('Association between Internet Usage Rate and Suicide Rate for those with HIGH income')

### Potential Moderator - ANOVA

In [ ]:
sub8['internetrate'] = sub8['internetrate'].convert_objects(convert_numeric=True)
sub9['internetrate'] = sub9['internetrate'].convert_objects(convert_numeric=True)

In [ ]:
print ('Association between Internet Usage Rates and Suicide Rates for those in LOW income countries')
model8 = smf.ols(formula='suicideper100th ~ C(internetrate)', data=sub8).fit()
print (model8.summary())

###Model Interpretation for ANOVA, LOW income:

When examining the association between suicide rate (quantitative response) and internet usage rate, an Analysis of Variance (ANOVA) revealed that suicide rate (Mean=9.7, s.d. ±6.3) was not related to Internet Usage Rate (Mean=33.97, s.d. ±27.7), F(1, 84)=.3319, p>.05.

print ('Association between Internet Usage Rates and Suicide Rates for those in HIGH income countries')
model9 = smf.ols(formula='suicideper100th ~ C(internetrate)', data=sub9).fit()
print (model9.summary())

###Model Interpretation for ANOVA, HIGH income:

When examining the association between suicide rate (quantitative response) and internet usage rate, an Analysis of Variance (ANOVA) revealed that suicide rate (Mean=9.7, s.d. ±6.3) was not related to Internet Usage Rate (Mean=33.97, s.d. ±27.7), F(1, 84)=.8566, p>.05.

####Compute Group Statistics

In [ ]:
sub10 = sub8[['suicideper100th', 'internetrate']]
print ('Means for Suicide Rates by Internet Use (LOW Income)')
m2= sub10.groupby('internetrate').mean()
print (m2)

In [ ]:
print ('Standard deviations for Suicide Rates by Internet Use (LOW Income)')
sd2 = sub10.groupby('internetrate').std()
print (sd2)

In [ ]:
sub11 = sub9[['suicideper100th', 'internetrate']]
print ('Means for Suicide Rates by Internet Use (HIGH Income)')
m2= sub11.groupby('internetrate').mean()
print (m2)

In [ ]:
print ('Standard deviations for Suicide Rates by Internet Use (HIGH Income)')
sd2 = sub11.groupby('internetrate').std()
print (sd2)

In [ ]:
print ("The minimum value for Internet use rate is: "), round(sub1["internetuserate"].min(),3)
print ("The maximum value for Internet use rate is: "), round(sub1["internetuserate"].max(),3)
print ("The average value for Internet use rate is: "), round(sub1["internetuserate"].mean(),3)
print ("The standard deviation for Internet use rate is: "), round(sub1["internetuserate"].std(),3)

In [ ]:
############################################################################################
# BASIC LINEAR REGRESSION USING INTERNET RATE
###########################################################################################
scat1 = seaborn.regplot(y="suicideper100th", x="internetuserate", scatter=True, data=sub1)
plt.xlabel('Internet Use Rate')
plt.ylabel('Suicide Rate')
plt.title ('Scatterplot for the Association Between Internet Use Rate and Suicide Rate')
print(scat1)

print ("OLS regression model for the association between Internet Use Rate and Suicide Rate")
reg1 = smf.ols('suicideper100th ~ internetuserate', data=sub1).fit()
print (reg1.summary())

The results of the linear regression model indicated that Internet Use Rate (Beta=0.0155, p=.373) was not significantly associated with suicide rates. Lets center Internet Usage Rates.

In [ ]:
fig1 = sm.qqplot(reg1.resid, line='r')

In [ ]:
# simple plot of residuals
stdres=pd.DataFrame(reg1.resid_pearson)
plt.plot(stdres, 'o', ls='None')
l = plt.axhline(y=0, color='r')
plt.ylabel('Standardized Residual')
plt.xlabel('Observation Number')

In [ ]:
print
internetuserate_c = sub1["internetuserate"]-sub1["internetuserate"].mean()
print ("The average value for Internet use rate (centered) is: "), round(internetuserate_c.mean(),2)

In [ ]:
############################################################################################
# BASIC LINEAR REGRESSION USING CENTERED INTERNET RATE
###########################################################################################
scat2 = seaborn.regplot(y="suicideper100th", x="internetuserate_c", scatter=True, data=sub1)
plt.xlabel('Internet Use Rate (Centered)')
plt.ylabel('Suicide Rate')
plt.title ('Scatterplot for the Association Between Internet Use Rate (Centered) and Suicide Rate')
print(scat2)

print ("OLS regression model for the association between Internet Use Rate (Centered) and Suicide Rate")
reg2 = smf.ols('suicideper100th ~ internetuserate_c', data=sub1).fit()
print (reg2.summary())

The results of the linear regression model indicated that Internet Use Rate (Beta=0.4275, p=.373) was not significantly associated with suicide rates.

In [ ]:
fig2= sm.qqplot(reg2.resid, line='r')

In [ ]:
# simple plot of residuals
stdres=pd.DataFrame(reg2.resid_pearson)
plt.plot(stdres, 'o', ls='None')
l = plt.axhline(y=0, color='r')
plt.ylabel('Standardized Residual')
plt.xlabel('Observation Number')

In [ ]:
###########################################################################################
# QUADRADIC REGRESSION USING CENTERED INTERNET RATE
###########################################################################################
scat3 = seaborn.regplot(y="suicideper100th", x="internetuserate_c", order=2, scatter=True, data=sub1)
plt.xlabel('Internet Use Rate (Centered)')
plt.ylabel('Suicide Rate')
plt.title ('Scatterplot for the Association Between Internet Use Rate (Centered) and Suicide Rate - Order=2')
print(scat3)

print ("OLS regression model for the association between Internet Use Rate (Centered) and Suicide Rate")
reg3 = smf.ols('suicideper100th ~ internetuserate_c+I(internetuserate_c**2)', data=sub1).fit()
print (reg3.summary())

In [ ]:
fig3 = sm.qqplot(reg3.resid, line='r')

In [ ]:
# simple plot of residuals
stdres=pd.DataFrame(reg3.resid_pearson)
plt.plot(stdres, 'o', ls='None')
l = plt.axhline(y=0, color='r')
plt.ylabel('Standardized Residual')
plt.xlabel('Observation Number')

# additional regression diagnostic plots
fig3a = plt.figure(figsize(12,8))
fig3a = sm.graphics.plot_regress_exog(reg3,  "internetuserate_c", fig=fig3a)

# leverage plot
fig3b=sm.graphics.influence_plot(reg3, size=8)
print(fig3b)

In [ ]:
fig3a = plt.figure(figsize(12,8))